synapse/storage/data_stores/main/room.py

```diff
@@ -98,6 +98,35 @@ def get_room(self, room_id):
             allow_none=True,
         )
 
+    def get_room_with_stats(self, room_id: str):
+        """Retrieve room with statistics.
+
+        Args:
+            room_id: The ID of the room to retrieve.
+        Returns:
+            A dict containing the room information, or None if the room is unknown.
+        """
+
+        def get_room_with_stats_txn(txn, room_id):
+            sql = """
+                SELECT room_id, state.name, state.canonical_alias, curr.joined_members,
+                  curr.local_users_in_room AS joined_local_members, rooms.room_version AS version,
+                  rooms.creator, state.encryption, state.is_federatable AS federatable,
+                  rooms.is_public AS public, state.join_rules, state.guest_access,
+                  state.history_visibility, curr.current_state_events AS state_events
+                FROM rooms
+                LEFT JOIN room_stats_state state USING (room_id)
+                LEFT JOIN room_stats_current curr USING (room_id)
+                WHERE room_id = ?
+                """
+            txn.execute(sql, [room_id])
+            res = self.db.cursor_to_dict(txn)[0]
+            res["federatable"] = bool(res["federatable"])
+            res["public"] = bool(res["public"])
```

Comment:
I like this! Maybe we can make the same change to the list API while we're at it? Right now that API returns ints for "federatable" and "public".

Reply:
Yeah, I can add it in this PR. Any idea why `_get_rooms_paginate_txn` doesn't use `self.db.cursor_to_dict` but instead builds the array itself?

Reply:
I do not know, but seems like @anoadragon453 added it! @anoadragon453 any idea if there's a reason not to use `cursor_to_dict` there?

Reply:
OK, let's do it in a separate PR and keep this one simple.